### PR TITLE
Add Python 3.14 support & a few small fixes and features

### PR DESCRIPTION
**Pull request by mbollmann**
- Adds Python 3.14 support.
- Updates all dependencies. New black major version (26.1) introduces minor styling changes.
- Refactors how "verbose" parameter works; among other things, this fixes a bug that produced empty lines when redirecting the output to a file. It also defaults to suppressing progress bars now if the stdout is redirected.
- Removes Papers with Code functionality, as we can assume that service is not coming back.
- Adds `Anthology.from_within_repo()`.
- Bumps version of this library for new release after the author system transition — I propose to call this v1.0.0 as it’s the first version that has complete save functionality, making it more or less feature-complete!

### CODE REVIEW

**Comment by mbollmann** [comment]
3.14 support is not functional yet because I can’t test it locally (need to upgrade my system first to get Python 3.14 🙃). I also plan to:

- Remove Papers with Code functionality, as we can assume that service is not coming back.
- Add `Anthology.from_within_repo()` or equivalent.
- Bump version of this library for new release after the author system transition.

**Comment by mbollmann** [comment]
@mjpost @nschneid I hope this has not gotten to bloated to review; I wanted to add Python 3.14 support and a few other quick things, plus bumping the version up to 1.0.0 — at least that’s my proposal because this feels more or less feature-complete for the first time. :)

My suggestion would be to merge -> `python-dev` -> `master-new-author-system` so that it goes on `master` at the same time as the author page transition, so I can immediately push a release to PyPI once that is live.

**Comment by mbollmann** [comment]
Note: If you agree on removing PwC functionality (I remember we said before we should give it until the end of 2025), I would make a separate PR to master to remove it from the Hugo templates as well.

**Comment by mjpost** [comment]
> Note: If you agree on removing PwC functionality (I remember we said before we should give it until the end of 2025), I would make a separate PR to master to remove it from the Hugo templates as well.

I agree on removing it. It's not coming back.

**Comment by mjpost** [comment]
Plan sounds fine, will review by tomorrow!

File: python/tests/collections/bibkeys_test.py
Comment (mjpost) [inline python/tests/collections/bibkeys_test.py:1]: Is there a test for the empty-title edge case you handled?

File: python/acl_anthology/utils/__init__.py [hunk 1]
```diff
@@ -25,7 +25,6 @@
 from .text import clean_unicode, remove_extra_whitespace
 from .xml import stringify_children
 
-
```
Comment (mjpost) [inline python/acl_anthology/utils/__init__.py:28]: Was this an explicit change to black, to remove the req for two blank lines at the top?

File: python/acl_anthology/anthology.py
Comment (mjpost) [inline python/acl_anthology/anthology.py:1]: One thing to think about, is that if you attempt to make changes after instantiating from the repo, it should throw an error

File: python/CHANGELOG.md [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
-## [Unreleased]
+## [1.0.0] — 2026-01-21
 
 This release implements the new [name resolution and author ID logic](https://github.com/acl-org/acl-anthology/wiki/Author-Page-Plan), and is therefore fundamentally incompatible with ACL Anthology data before the switch to this new system.
 
 ### Added
 
-- Anthology now provides `save_all()` to conveniently save all data files.  The library tracks modifications to collection objects to only write XML files that have actually changed.
+- Support for Python 3.14.
+- Anthology now provides `save_all()` to conveniently save all data files.  The library tracks modifications to collection objects to only write XML files that have actually changed.  (Tracking changes does not work on _every_ possible modification, though; see the documentation.)
```
Comment (mjpost) [inline python/CHANGELOG.md:10]: See [this comment](https://github.com/acl-org/acl-anthology/pull/6859#discussion_r2640505358)

File: python/acl_anthology/utils/__init__.py (hunk 1 above)
Comment (mbollmann) [reply python/acl_anthology/utils/__init__.py:28]: Yes, it's the first item in the change log
https://black.readthedocs.io/en/stable/change_log.html

File: python/tests/collections/bibkeys_test.py
Comment (mbollmann) [reply python/tests/collections/bibkeys_test.py:1]: Yes, the one with all stopwords, because it causes the title to be empty during bibkey generation.

Comment (mbollmann) [reply python/tests/collections/bibkeys_test.py:1]: Side note: Codecov would complain if there wasn't ;)

**Comment by mbollmann** [comment]
> LGTM. The only other thing I can think of is that it would be nice to also close some other issues (e.g., #6825 #7132 #7183 #7184). Not sure if it would make sense to include here.

Those are all addressed by 72fddcf7bab34ce69cfa01cf42ece06eafefd8fd.

File: python/acl_anthology/anthology.py
Comment (mbollmann) [reply python/acl_anthology/anthology.py:1]: Pushed an attempt to implement that
